fuel-core/src/database/coin.rs

```diff
@@ -122,7 +124,7 @@ impl Database {
         direction: Option<IterDirection>,
     ) -> impl Iterator<Item = Result<UtxoId, Error>> + '_ {
```

Comment:
Maybe for a separate PR, seems like we should return the actual `Coin` since we do the work of fetching it from storage to check the spent status anyways:

```suggestion
    ) -> impl Iterator<Item = Result<Coin, Error>> + '_ {
```

Reply:
It is already done in the https://github.com/FuelLabs/fuel-core/pull/591 =)